google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleFrameworkDetector.java

```diff
@@ -18,24 +18,34 @@
 
 import com.google.cloud.tools.intellij.stats.UsageTrackerProvider;
 import com.google.cloud.tools.intellij.util.GctTracking;
-
 import com.intellij.facet.FacetType;
 import com.intellij.framework.detection.FacetBasedFrameworkDetector;
-import com.intellij.framework.detection.FileContentPattern;
 import com.intellij.openapi.fileTypes.FileType;
 import com.intellij.openapi.roots.ModifiableRootModel;
 import com.intellij.patterns.ElementPattern;
-import com.intellij.patterns.StandardPatterns;
+import com.intellij.patterns.ObjectPattern;
+import com.intellij.patterns.PatternCondition;
+import com.intellij.util.ProcessingContext;
 import com.intellij.util.indexing.FileContent;
-
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.yaml.YAMLFileType;
 
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Collection;
+import java.util.Scanner;
+
 /** Detects App Engine Flexible framework in a project. */
 public class AppEngineFlexibleFrameworkDetector
     extends FacetBasedFrameworkDetector<
         AppEngineFlexibleFacet, AppEngineFlexibleFacetConfiguration> {
 
+  // app engine flex file names.
+  static final Collection<String> APP_ENGINE_FLEX_PROJECT_FILES =
+      Collections.unmodifiableList(Arrays.asList("app.yaml", "app.yml"));
+  // required string in app engine flex YAML project file.
+  static final String APP_ENGINE_REQUIRED_YAML = "runtime:";
```

Comment:
APP_ENGINE_FLEX_REQUIRED_YAML_CONTENT? Just to keep consistent with your pattern method naming